sdk/src/asset_handlers/otf_io.rs

```diff
@@ -298,6 +300,151 @@ pub enum FontVersion {
     AppleTrue = 0x74727565,
 }
 
+/// Declares the type of chunks of data within a font
+#[derive(Debug, PartialEq, Eq)]
+pub enum ChunkType {
+    /// Table directory, excluding the table record array
+    TableDirectory,
+    /// Table data
+    Table,
+    /// Table record entry in the table directory
+    TableRecord,
+}
+
+/// Represents positions within a font file that may be of interest when it
+/// comes to hashing data for C2PA
+#[derive(Debug)]
+pub struct SfntChunkPositions {
+    /// Offset to the start of the chunk
+    pub offset: u64,
+    /// Length of the chunk
+    pub length: u32,
+    /// Tag of the chunk
+    pub name: [u8; 4],
+    /// Type of chunk
+    pub chunk_type: ChunkType,
+}
+
+/// Custom trait for reading chunks of data from a scalable font (SFNT).
+pub trait SfntChunkReader {
+    type Error;
+    /// Gets a collection of positions of chunks within the font.
+    ///
+    /// ## Arguments
+    /// * `source_stream` - Source stream to read data from
+    ///
+    /// ## Returns
+    /// A collection of positions/offsets and length to omit from hashing.
+    fn get_chunk_positions<T: Read + Seek + ?Sized>(
+        &self,
+        source_stream: &mut T,
+    ) -> core::result::Result<Vec<SfntChunkPositions>, Self::Error>;
+}
+
+impl SfntChunkReader for OtfIO {
+    type Error = crate::error::Error;
+
+    fn get_chunk_positions<T: Read + Seek + ?Sized>(
+        &self,
+        source_stream: &mut T,
+    ) -> core::result::Result<Vec<SfntChunkPositions>, Self::Error> {
+        source_stream.rewind()?;
+        let mut positions: Vec<SfntChunkPositions> = Vec::new();
+        let table_header_sz: u32 = 12;
+        let table_entry_sz: u32 = 16;
+        // Create a 16-byte buffer to hold each table entry as we read through the file
+        let mut table_entry_buf: [u8; 16] = [0; 16];
+        // We need to get the offset to the 'name' table and exclude the length of it and its data.
+        // Verify the font has a valid version in it before assuming the rest is
+        // valid (NOTE: we don't actually do anything with it, just as a safety check).
+        let sfnt_u32: u32 = source_stream.read_u32::<BigEndian>()?;
+        let _sfnt_version: FontVersion =
+            <u32 as std::convert::TryInto<FontVersion>>::try_into(sfnt_u32)
+                .map_err(|_err| Error::UnsupportedFontError)?;
+
+        // Add the position of the table directory, excluding the actual table
+        // records, as those positions will be added separately
+        positions.push(SfntChunkPositions {
+            offset: 0,
+            length: 12,
```

Comment:
Nit: Would it be worth defining a constant for the `12`?  `TableDirectoryHeader` (as that's _kind_ of what "table directory without tables" is?)  You use it down below when doing the seek too